packages/mdc-chips/chip/foundation.ts

```diff
@@ -334,9 +334,7 @@ export class MDCChipFoundation extends MDCFoundation<MDCChipAdapter> {
       this.adapter_.setPrimaryActionAttr(strings.ARIA_CHECKED, 'false');
     }
 
-    if (shouldNotify) {
-      this.adapter_.notifySelection(selected);
-    }
+    this.adapter_.notifySelection(selected, shouldIgnore);
```

Comment:
Instead of introducing a boolean flag to decide whether to emit an event, can we modify foundation to notify only when needed? I don't see why we needed this.

Reply:
It was this way, however I think @patrickrodee had issues with this design, correct?

Reply:
That's correct. We have existing cases in Google that require the selection event to be emitted when the selection changes.

Reply:
I might be lacking some context here. At least not of enough context provided in this PR about new boolean option.

Here is my understanding, `setSelectedFromChipSet` method is invoked by chip set (parent). When this method invoked by chip set, the chip doesn't need to notify the selection event to avoid infinite loop.

My suggestion was that, `setSelected_` private method can be split into two separate private methods, one for presentational only changes (`setSelected_`) and another for notifying event (`notifySelection_`). If we do this, `setSelected` calls `setSelected_` & `notifySelection_` where `setSelectedFromChipSet` just calls `setSelected_` (presentational only changes).

WDYT?

Reply:
I've update the functionality. The chip set is now conditionally able to block events from bubbling and also label events as ignorable.

Reply:
Also updated to use the suggestion above.